R/FilteredDataset.R

```diff
@@ -42,6 +42,11 @@ FilteredDataset <- R6::R6Class( # nolint
       private$dataname <- dataname
       private$keys <- keys
       private$label <- if (is.null(label)) character(0) else label
+      private$reactive_call <- reactiveVal()
+
+      observeEvent(self$get_call(), ignoreNULL = FALSE, {
```

Comment:
this is constructor - it should be outside of a reactive context. It is only because `srv_teal` `insertUI` which is wrong and will be removed sooner or later.

Reply:
Reverted that change 👍🏽 
I was surprised that observing outside a module server worked just fine. Right now the observer is inside the `srv_active` which is also not correct. It should be somewhere similar to the global server init. I will try to move this to some server logic in the `FilterData` class.
But, this adds a new problem. The dataset is only dependant on the `private$reactive_call ()` which needs shiny to work. So, we have some unit tests failing. As, those tests used to be reactive on the `self$get_call()` before.